schema/global/people.ts

```diff
@@ -36,3 +37,93 @@ export const PersonQuery = extendType({
   }
   // TODO: filter out nilPersonId
 })
+
+export const DirectoryPerson = objectType({
+	name: "DirectoryPerson",
+	definition(t) {
+		t.string("name")
+		t.string("surname")
+		t.string("email")
+		t.int("sciper")
+		t.string("type")
+	}
+})
+
+export const DirectoryOrLhdPerson = unionType({
+	name: "DirectoryOrLhdPerson",
+	definition(t) {
+		t.members("Person", "DirectoryPerson");
+	},
+	resolveType(item) { return "id_person" in item ? "Person" : "DirectoryPerson"}
+});
+
+export const PersonFullTextQuery = extendType({
+	type: 'Query',
+	definition(t) {
+		t.field("personFullText", {
+			type: list("DirectoryOrLhdPerson"),
+			args: {
+				search: stringArg(),
+				lhdOnly: booleanArg()
+			},
+			async resolve(parent, args, context) {
+
+				const lhdPeople = await context.prisma.Person.findMany({
+					where: {
+						OR: [
+							{ name: { contains: args.search }},
+							{ surname : { contains: args.search }},
+						]
+					}
+				});
+				const lhdPeopleTyped = lhdPeople.map(p => {
+					return {
+						type: 'Person',
+						name: p.name,
+						surname: p.surname,
+						email: p.email,
+						sciper: p.sciper
+					}
+				});
+
+
+				let ldapUsers = [];
+				if (!args.lhdOnly) {
+					ldapUsers = await getUsers(args.search).then(users => {
+						const result = [];
+						users.forEach(u => {
+							if (!lhdPeopleTyped.find(p => p.sciper == u.sciper)) {
+								result.push({
+									type: 'DirectoryPerson',
+									surname: u.name,
+									name: u.firstname,
+									email: u.email,
+									sciper: u.sciper
+								});
+							}
+						});
+						return result;
+					});
+				}
+				return lhdPeopleTyped.concat(ldapUsers);
+			}
+		})
+	},
+})
+
+function getUsers(search: string): Promise<any[]> {
```

Comment:
Please rename to `getUsersSearchApi`. Please make it an async function.

Reply:
Done